main/lsp/ErrorReporter.cc

```diff
@@ -155,7 +155,15 @@ void ErrorReporter::pushDiagnostics(uint32_t epoch, core::FileRef file, const ve
             tags.push_back(DiagnosticTag::Unnecessary);
             diagnostic->tags = move(tags);
         }
+
         diagnostic->severity = DiagnosticSeverity::Error;
+        if (error->what == sorbet::core::errors::Infer::UntypedValueInformation) {
+            if (config->getClientConfig().enableWarnUntypedValues) {
```

Comment:
ah, i misunderstood how this feature worked.

It _should_ be redundant, but it's not—this check needs to happen earlier, in inference. You haven't been able to do that, because you've noticed that you don't have `getClientConfig()` in the pipeline. That's intentional—you need to store this on GlobalState. Look at another of the boolean property assertions for how to get this onto `GlobalState &`

The point of the `beginError` functionality is that the logic inside the body of the `if` doesn't run if the error wouldn't be used anyways. Right now what happens is that Sorbet does all the work of building and flushing these errors, but simply _drops them on the floor_ in this function. That's too late—we should not have to `continue` in this function, rather the error should have never been reported in the first place.